EdFi.Buzz.UI.Angular/src/app/Features/UploadSurvey/uploadSurvey.tsx

```diff
@@ -0,0 +1,415 @@
+        </div>
+
+        {surveyToUpdate && (
+          <div className='row'>
+            <div className='col'>
+              <div className='card'>
+                <div className='card-body'>
+                  <h2>You are currently updating the {surveyToUpdate.resultSummaryObj.survey.title} survey</h2>
+                </div>
+              </div>
+            </div>
+          </div>
+        )
+        }
+        <div className='row'>
+          <div className='col'>
+            <div className='card'>
+              <div className='card-body'>
+                <form onSubmit={submitSurvey}>
+                  <div className='container text-center p-b-10'
+                    onDrop={onFileDropped}
+                    onDragLeave={ev => ev.preventDefault()}
+                    onDragEnter={ev => ev.preventDefault()}
+                    onDragOverCapture={onDragOver} onDragOver={onDragOver}>
+                    <p className='icon ion-md-cloud-upload f-s-50'></p>
+                    <h3>Drag and drop file here</h3>
+                    <h3>or</h3>
+                    <div className='custom-file'>
+                      <input type='file' onChange={onSelectSurvey}
+                        disabled={isFileUploading} className='custom-file-input' id='inputGroupFile01'
+                        accept='text/csv' />
+                      <label className='custom-file-label'>{uploadFileLabelText}</label>
+                    </div>
+                  </div>
+                  <div className='form-group'>
+                    <input type='text' className='form-control' placeholder='Survey Name' onChange={setSurveyTitle} value={surveyName} />
+                    {(surveyName.length === 0 && isFileValid)
+                      ? <label className='label alert-danger text-justify w-100'> * Survey name is required </label>
+                      : null}
+                  </div>
+                  <div className='form-group'>
+                    <button type='submit' className='btn btn-primary btn-block btn-lg'
+                      disabled={!isFormValid}>Upload survey</button>
+                  </div>
+                  {(progress > 0)
+                    ?
+                    <div className='progress form-group'>{// TODO Progress bar style.width.%={progress}
+                    }
+                      <div className='progress-bar progress-bar-striped bg-success' role='progressbar'>
+                      </div>
+                    </div>
+                    : null
+                  }
+                  {fileStatusMessage
+                    && fileStatusMessage.fileName
+                    && fileStatusMessage.fileName.length > 0
+                    ?
+                    <div
+                      className={(fileStatusMessage
+                        && fileStatusMessage.error
+                        || (fileStatusMessage.serverJobStatus
+                          && fileStatusMessage.serverJobStatus.jobstatuskey !== 3))
```

Comment:
Consider a [use effect hook here](https://daveceddia.com/useeffect-hook-examples/#:~:text=Run%20useEffect%20on%20State%20Change,the%20second%20argument%20to%20useEffect.) to consolidate this check whenever fileStatusMessage changes.


```javascript
useEffect(() => {
/* ... all of this code that sets the className*/
}, [fileStatusMessage]);

```